src/scripts/main.js

```diff
@@ -18,10 +18,34 @@ const food = {
   },
 };
 
+const treeDiv = document.createElement('div');
+
+treeDiv.id = 'tree';
+document.body.appendChild(treeDiv);
+
 const tree = document.querySelector('#tree');
 
 function createTree(element, data) {
-  // WRITE YOUR CODE HERE
+  if (element === null || data === undefined) {
+    throw new Error('arguments must not be null or undefined');
+  }
+
+  const ul = document.createElement('ul');
+
+  for (const key in data) {
+    const li = document.createElement('li');
+
+    li.textContent = key;
+
+    if (Object.entries(data[key]).length > 0) {
+      createTree(li, data[key]);
+    }
+    ul.appendChild(li);
+  }
+
+  if (ul.childNodes.length > 0) {
+    element.appendChild(ul);
+  }
```

Comment:
It's good that you are checking if the list has any items before appending it to the element. However, this check is not necessary because the `appendChild` method won't do anything if the list is empty. You can safely remove this check.